web/main/admin.py

```diff
@@ -45,16 +41,18 @@
 
 
 class FasterAdminPaginator(Paginator):
+    """Does pagination estimation only if the total number of rows is high and there are no filters applied"""
+
     @cached_property
     def count(self) -> int:
         cursor = connection.cursor()
         cursor.execute(
             f"SELECT reltuples AS estimate FROM pg_class WHERE relname = '{self.object_list.query.model._meta.db_table}';"
         )
         estimate = int(cursor.fetchone()[0])
-        if estimate > 10000:
+        if estimate > 10_000 and not bool(self.object_list.query.where):
```

Comment:
Nice! If it turns out this works, I might see if we can do similarly in Perma...